WordPress/playground-tools
# Add WordPress Playground plugin

## What?

This adds the `collector` plugin for plugin-preview functionality and renames the plugin to WordPress Playground.

Requires:
https://github.com/WordPress/wordpress-playground/pull/745 .
https://github.com/WordPress/blueprints/issues/49

## Why?

Users want to be able to preview plugins on their own sites before actually committing to installing them live.

## Testing Instructions

- Checkout the project and copy the plugin to your local WordPress install:
```bash
git checkout sm-collector-plugin
cp packages/collector/ [PATH_TO_YOUR_WORDPRESS]/wp-content/plugins/
```
- Open your local WordPress install and activate the WordPress Playground plugin
- Start Playground by clicking on Tools > Sandbox Site in wp-admin
- After the site loads it should have the same content as your site (files and database)

### Testing Previews
- Open the file at `[PATH_TO_YOUR_WORDPRESS]/wp-content/plugins/collector/Collector.php`. Replace line 81 in line `Collector.php` with:
```php
const blueprintUrl = (query.get('blueprintUrl') ?? '').replace('https://wordpress.org', 'http://localhost:8010/proxy');
```
- Start the CORS proxy:
```bash
npm install -g local-cors-proxy
npx lcp --proxyUrl https://wordpress.org/
```
- Navigate to `plugins` > `add new`
- Search for "Akismet"
- You should now see a `Preview Now` button next to the `Install Now` button.
- Click `Preview now`, and you should see Playground boot your site with the plugin activated.
